### ISSUE TITLE
Lookup window not showing correct glyph for the selected language

### ISSUE THREAD

**Issue by brainwo** [ibus/ibus#2677]
**Which distribution and version?:** 
Arch Linux

**Which desktop environment and version?:** 
Not correct ~~Inapplicable, reproducible in any desktop environment.~~
 XFCE and BSPWM

**Which session type?:** 
Not correct ~~Inapplicable.~~
Xorg

**Which application and version?:** 
Inapplicable, reproducible in any application.

**IBus version?:** 
IBus 1.5.30

**Issue description:**
Lookup window doesn't use the correct Han character for Japanese unless font is set to a Japanese font, but that would set Japanese font for Chinese IME.

**Steps to reproduce:**
1. Enable "Choose glyphs with input method's language on candidate window" option in the Advanced tab.
1. Uncheck the "Use custom font:" option in the General tab.
1. Switch to "Japanese - Anthy" input method.
1. Type and find any [Unified Han Characters](https://en.wikipedia.org/wiki/Han_unification) (e.g. "nao/なお").
1. The Chinese version of the character will be shown instead of the Japanese version.
![image](https://github.com/user-attachments/assets/42e8d0a1-66ce-4e8c-a2b2-7e440b275ef9)
![image](https://github.com/user-attachments/assets/4028266a-090f-45db-ad04-35438ea4ad58)
1. Now check the "Use custom font:" and pick a Japanese font. (e.g. Source Han Sans JP Regular).
1. Repeat the (3) and (4) step, the character will be shown as intended (Japanese version for <span lang="ja">直</span>).
1. Switch to "Chinese - Rime" input method.
1. Type and find any [Unified Han Characters](https://en.wikipedia.org/wiki/Han_unification) (e.g. "zhi").
1. The Japanese version of the character will be shown instead of the Chinese version.
![image](https://github.com/user-attachments/assets/f77a33b3-fb4a-48cb-8eb2-878120283d97)
1. This results in frustration to check and uncheck "Use custom font:" option to get the correct character in the corresponding IME.

**Can you reproduce your problem when you restart ibus-daemon? (yes / no):** 
Yes

**Do you see any errors when you run ibus-daemon with the verbose option?:**
There are no errors appear when step is being reproduced, however there are these warning when starting ibus:
```
(ibus-daemon:171711): GLib-CRITICAL **: 16:20:20.055: g_variant_new_string: assertion 'string != NULL' failed

(ibus-ui-gtk3:171722): IBUS-WARNING **: 16:20:20.155: panel.vala:340: If you launch KDE5 on xterm, export XDG_CURRENT_DESKTOP=KDE before launch KDE5.

(ibus-ui-gtk3:171722): IBUS-WARNING **: 16:20:20.175: ibus_bus_call_sync: org.freedesktop.DBus.Properties.Get: GDBus.Error:org.freedesktop.DBus.Error.Failed: No global engine.

(ibus-ui-gtk3:171722): Gdk-CRITICAL **: 16:20:20.250: gdk_window_thaw_toplevel_updates: assertion 'window->update_and_descendants_freeze_count > 0' failed
```

**Can you reproduce your problem with a new user account instead of the current your account? (yes / no):** 
Yes

---

# Original Issue:

Currently it's only possible to set a custom font for all IBus input methods.

![image](https://github.com/user-attachments/assets/9f8cd52d-ee55-4ec2-b6c8-ddb38b283f58)

This limitation wouldn't be good for user of multiple CJK languages. I don't understand exactly how these fonts nor how Han unification should work, but it seems like IBus can't figure out the correct locale when using a non specified font. Regional characters aren't showed correctly, it become tedious to change custom font every time I change my input method.

While unified characters aren't that many to be remembered, being able to see the correct regional character would be great. Ideally, IBus should allow customizing font for each individual input method.

**Comment by fujiwarat** [comment]
You should use the bug template. I don't understand your issue and please provide the  steps to reproduce as the bug template suggests otherwise I will close this issue.
I guess your issue could be resolved with "Choose glyphs with input method's language on candidate window" in Advanced tab.

**Comment by brainwo** [comment]
> You should use the bug template.

I apologize for ignoring the issue template. I thought it was inapplicable to the problem I have, so I went straight up writing a feature request issue. I will update the original issue using the template.

---

>  I guess your issue could be resolved with "Choose glyphs with input method's language on candidate window" in Advanced tab.

I do have "Choose glyph with input method's language on candidate window" option enabled. I don't think it helps with anything.

![image](https://github.com/user-attachments/assets/1a1f9b78-58ca-4d3c-863b-0e9842264090)

---

>  I don't understand your issue and please provide the steps to reproduce as the bug template suggests otherwise I will close this issue.

_(This part will also be updated in the issue)_

## System informations:
- Output of `locale`
```
LANG=en_US.utf8
LC_CTYPE="en_US.utf8"
LC_NUMERIC="en_US.utf8"
LC_TIME="en_US.utf8"
LC_COLLATE="en_US.utf8"
LC_MONETARY="en_US.utf8"
LC_MESSAGES="en_US.utf8"
LC_PAPER="en_US.utf8"
LC_NAME="en_US.utf8"
LC_ADDRESS="en_US.utf8"
LC_TELEPHONE="en_US.utf8"
LC_MEASUREMENT="en_US.utf8"
LC_IDENTIFICATION="en_US.utf8"
LC_ALL=
```
- Installed font packages: `noto-fonts-cjk`, `adobe-source-han-sans-jp-fonts`

## Step to reproduce:
1. Enable "Choose glyphs with input method's language on candidate window" option in the Advanced tab.
1. Uncheck the "Use custom font:" option in the General tab.
1. Switch to "Japanese - Anthy" input method.
1. Type and find any [Unified Han Characters](https://en.wikipedia.org/wiki/Han_unification) (e.g. "nao/なお").
1. The Chinese version of the character will be shown instead of the Japanese version.
![image](https://github.com/user-attachments/assets/42e8d0a1-66ce-4e8c-a2b2-7e440b275ef9)
![image](https://github.com/user-attachments/assets/4028266a-090f-45db-ad04-35438ea4ad58)
1. Now check the "Use custom font:" and pick a Japanese font. (e.g. Source Han Sans JP Regular).
1. Repeat the (3) and (4) step, the character will be shown as intended (Japanese version for <span lang="ja">直</span>).
1. Switch to "Chinese - Rime" input method.
1. Type and find any [Unified Han Characters](https://en.wikipedia.org/wiki/Han_unification) (e.g. "zhi").
1. The Japanese version of the character will be shown instead of the Chinese version.
![image](https://github.com/user-attachments/assets/f77a33b3-fb4a-48cb-8eb2-878120283d97)
1. This results in frustration to check and uncheck "Use custom font:" option to get the correct character in the corresponding IME.

**Comment by fujiwarat** [comment]
> 1. Enable "Choose glyphs with input method's language on candidate window" option in the Advanced tab.
> 2. Uncheck the "Use custom font:" option in the General tab.
> 3. Switch to "Japanese - Anthy" input method.
> 4. Type and find any [Unified Han Characters](https://en.wikipedia.org/wiki/Han_unification) (e.g. "nao/なお").
> 5. The Chinese version of the character will be shown instead of the Japanese version.

Hmm.., your result mean "Choose glyphs with input method's language..." does not work. I will check the behavior later.

> Which desktop environment and version?:
> Inapplicable, reproducible in any desktop environment.

Which desktop environment do you use? XFCE?
It's implemented for non-GNOME desktop environment.

**Comment by brainwo** [comment]
I use XFCE and BSPWM. In BSPWM, I use mostly GTK applications but I don't use it with GNOME shell, if that helps with anything.

**Comment by fujiwarat** [comment]
> Can you reproduce your problem with a new user account instead of the current your account? (yes / no):
> Yes

Really??

**Comment by brainwo** [comment]
I found the problem. I will close this issue. The problem is on my end with the font I'm using. Apparently I should be using `Source Han`. I admit I don't understand fully how these fonts actually work. I was very confused when you said it supposed to work without any additional setup, yet I tried doing fresh installation and still run to the same issue.

Thank you for your patience in replying to this issue, I highly appreciate it. 🙇‍♂️

---

Below is reply for the questions:

>  you don't describe your issue exactly.

I'm sorry if I don't explain my problem well enough. On default font, the candidate window shows Chinese version of the Han character while using Japanese input method despite of having the "Choose glyphs with input method's language on candidate window" option enabled.

> You should provide tried applications exactly.

I tried on Mousepad, Discord, and Firefox. I have `GTK_IM_MODULE` and other equivalent variables all set to `ibus`.

> Which desktop fonts are configured in your XFCE?

It seems like the Arch Linux build of XFCE package is using a font called `Sans`, I'm not really sure what this font is. But I'm guessing the font only comes with Chinese variant of the characters, while Fedora comes with a font similar to `Source Han`.

I was also using `Noto Sans CJK` as I showed in my previous screenshot. But apparently I shouldn't be using a region-specific font. I'm confused with seeing `Noto Sans CJK JP`, `Noto Sans CJK TC`, and so on are available.

> Really??

A very big yes. I was very confused and had to double check this.

Below is the recording of my latest attempt. (It was using the default system font from XFCE).

https://github.com/user-attachments/assets/963676d3-b14b-4a91-9513-bca8749ba6cb

As I mentioned above, I have resolved this issue using `Source Han Sans` font. Thank you again for reading this.
